composition/composition.md

```diff
@@ -151,30 +159,63 @@ Now, I'm going to leave Component's instance method
 This is what Component's method returned: 473
 ```
 
-## A Longer Example: Product and ShoppingCart
+## One-to-One and One-to-Many Composition Relationships
+
+We just saw examples of one-to-one and one-to-many composition relationships.
+
+A one-to-one relationship is one where a composite class has an attribute single instance of a component class as in this example:
 
 ```python
-class Product:
+class ExampleComponent:
 
-    def __init__(self, price, quantity):
-        self.price = price
-        self.quantity = quantity
+    def __init__(self, name):
+        self.name = name
 
-    def calculate_cost(self):
-        return self.price * self.quantity
+class ExampleComposite:
 
-class ShoppingCart:
+    def __init__(self, name, component):
+        self.name = name
+        self.component = component
+```
 
-    def __init__(self, products):
-        self.products = products
+A one-to-many relationship is one where a composite class has an attribute that is a list of instances of a component class as in this example:
 
-    def calculate_total_cost(self):
-        total_price = 0.0
+```python
+class ExampleComposite:
 
-        for product in self.products:
-            total_price += product.calculate_cost()
+    def __init__(self, name, components):
+        self.name = name
+        self.components = components
+```
+
+Note that `components` is a list of `component` objects.
+
+## A Longer Example: Product and ShoppingCart
+
+Now we will look at one more example of a one-to-many composition relationship.
 
-        return total_price
+```python
+1 class Product:
+2 
+3    def __init__(self, price, quantity):
+4        self.price = price
+5        self.quantity = quantity
+6
+7    def calculate_cost(self):
+8        return self.price * self.quantity
+9
```

Comment:
consider adding spaces to keep things in alignment. Alternatively, what about using 01, 02, 03, ...
```suggestion
1  class Product:
2 
3     def __init__(self, price, quantity):
4         self.price = price
5         self.quantity = quantity
6
7     def calculate_cost(self):
8         return self.price * self.quantity
9
```